File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
# 프로그래머스 42746 가장 큰 수 LV.2
def solution(numbers):
    answer = ''
    numbers = sorted(map(str, numbers), key=lambda x: x * 3, reverse=True)
    for n in numbers:
        if answer == '' and n == '0':
            return '0'
        answer += n
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
        Integer[] nums = Arrays.stream(numbers).boxed().toArray(Integer[]::new);
        Arrays.sort(nums, (o1, o2) -> {
            String str1 = String.valueOf(o1), str2 = String.valueOf(o2);
            return (str2 + str1).compareTo(str1 + str2);
        });
                
        StringBuilder sb = new StringBuilder();
        for (int n : nums) sb.append(n);
        return sb.toString().startsWith("0") ? "0" : sb.toString();
    }
}